### ISSUE TITLE
[defect]: python3 module crashes freeradius

### ISSUE THREAD

**Issue by Sprinterfreak** [FreeRADIUS/freeradius-server#4953]
### What type of defect/bug is this?

Crash or memory corruption (segv, abort, etc...)

### How can the issue be reproduced?

* Install freeradius-python3 from debian bullseye repositories, enable python3 module
* Install freeradius 3.0.26-1 from networkradius repo's, enable python3 module
* Build freeradius 3.0.27 from source, enable python3 module

### Log output from the FreeRADIUS daemon

```shell
see #4951
```

### Relevant log output from client utilities

Does not apply, unable to start server

### Backtrace from LLDB or GDB

```shell
Installed from debian upstream see #4951 for available trackback

networkradius and built from source do not provide library rlm_python3.so (No such file or directory)
```

**Comment by Sprinterfreak** [comment]
Hi,

https://github.com/FreeRADIUS/freeradius-server/issues/4952#issuecomment-1493184328
@arr2036 thanks for the answer again.

Information regarding the backtrace: Yes, it's not much information but it is complete console output from required calls, stated in the issue template. Yes, it crashes that hard, that quickly.
Only the second startup inside the debugger is shortened, because it is an identical duplicate of console output from the above call outside the debugger. I'm by all means no C or low-level developer, so I'm not familiar with deep lldb debugging. Just be proud that I knew how to run the application inside.

To the debian packaging, and I tried that twice 5 mins ago.
The debian bullseye stable repo provides a package called `freeradius-python3` which contains rlm_python3.so.
The networkradius repo (neither 3.0/3.2) does provide a similar package. Installing `apt install freeradius` from networkradius repo's does not provide rlm_python3.so.
Thats why in my next post there is the startup log provided:
```
Apr 02 01:51:27 freerad freeradius[50888]: FreeRADIUS Version 3.0.26
Apr 02 01:51:27 freerad freeradius[50888]: Copyright (C) 1999-2021 The FreeRADIUS server project and contributors
Apr 02 01:51:27 freerad freeradius[50888]: There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
Apr 02 01:51:27 freerad freeradius[50888]: PARTICULAR PURPOSE
Apr 02 01:51:27 freerad freeradius[50888]: You may redistribute copies of FreeRADIUS under the terms of the
Apr 02 01:51:27 freerad freeradius[50888]: GNU General Public License
Apr 02 01:51:27 freerad freeradius[50888]: For more information about these matters, see the file named COPYRIGHT
Apr 02 01:51:27 freerad freeradius[50888]: Starting - reading configuration files ...
Apr 02 01:51:27 freerad freeradius[50888]: Debug state unknown (cap_sys_ptrace capability not set)
Apr 02 01:51:27 freerad freeradius[50888]: /etc/freeradius/mods-enabled/python3[9]: Failed to link to module 'rlm_python3': /usr/lib/freeradius/rlm_python3.so: cannot open shared object file: No such file or directory
Apr 02 01:51:27 freerad systemd[1]: freeradius.service: Control process exited, code=exited, status=1/FAILURE
Apr 02 01:51:27 freerad systemd[1]: freeradius.service: Failed with result 'exit-code'.
Apr 02 01:51:27 freerad systemd[1]: Failed to start FreeRADIUS multi-protocol policy server.
```
freeradius only installs dependencies python2.7 which has no use for me. rlm_python3.so: "No such file or directory", reported by a clean freeradius 3.0.26-1 install from networkradius repo's and just python3 enabled by linking the example `python3`-file from mods-available to -enabled. So looks pretty hard for me to further test.
I also manually looked into freeradius_3.0.26-1_amd64.deb. There is no rlm_python3.so included.

Also looks like the issue is not new: https://lists.freeradius.org/pipermail/freeradius-users/2021-November/101085.html
Despite rewriting code to work with python2.7 is not an option for me.

If there is more too look for, please request for more specific e.g. parameters to try. 

```
# apt search freeradius*
Sorting... Done
Full Text Search... Done
argonaut-freeradius/stable 1.3-2 all
  Argonaut scripts to generate authentication for FreeRADIUS

freeradius/bullseye,now 3.0.26-1 amd64 [installed]
  high-performance and highly configurable RADIUS server

freeradius-common/bullseye,now 3.0.26-1 all [installed,automatic]
  FreeRADIUS common files

freeradius-config/bullseye,now 3.0.26-1 amd64 [installed,automatic]
  FreeRADIUS default config files

freeradius-dbg/bullseye 3.0.26-1 amd64
  debug symbols for the FreeRADIUS packages

freeradius-dhcp/bullseye 3.0.26-1 amd64
  DHCP module for FreeRADIUS server

freeradius-freetds/bullseye 3.0.26-1 amd64
  FreeTDS support for FreeRADIUS

freeradius-iodbc/bullseye 3.0.26-1 amd64
  iODBC module for FreeRADIUS server

freeradius-krb5/bullseye 3.0.26-1 amd64
  kerberos module for FreeRADIUS server

freeradius-ldap/bullseye 3.0.26-1 amd64
  LDAP module for FreeRADIUS server

freeradius-memcached/bullseye 3.0.26-1 amd64
  Memcached module for FreeRADIUS server

freeradius-mysql/bullseye 3.0.26-1 amd64
  MySQL module for FreeRADIUS server

freeradius-perl-util/bullseye 3.0.26-1 amd64
  FreeRADIUS Perl utilities

freeradius-postgresql/bullseye 3.0.26-1 amd64
  PostgreSQL module for FreeRADIUS server

freeradius-python3/stable 3.0.21+dfsg-2.2+deb11u1 amd64
  Python 3 module for FreeRADIUS server

freeradius-redis/bullseye 3.0.26-1 amd64
  Redis module for FreeRADIUS server

freeradius-rest/bullseye 3.0.26-1 amd64
  REST module for FreeRADIUS server

freeradius-utils/bullseye,now 3.0.26-1 amd64 [installed,automatic]
  FreeRADIUS client utilities

freeradius-yubikey/bullseye 3.0.26-1 amd64
  Yubikey module for FreeRADIUS server

fusiondirectory-plugin-freeradius/stable 1.3-4 all
  freeradius plugin for FusionDirectory

fusiondirectory-plugin-freeradius-schema/stable 1.3-4 all
  LDAP schema for FusionDirectory freeradius plugin

libfreeradius-dev/bullseye 3.0.26-1 amd64
  FreeRADIUS shared library development files

libfreeradius3/bullseye,now 3.0.26-1 amd64 [installed,automatic]
  FreeRADIUS shared library

libradcli4/stable 1.2.11-1+b2 amd64
  Enhanced RADIUS client library
```

See here, the freeradius-python3 package is not provided by the networkradius repo. It is still listed version 3.0.21+dfsg-2.2+deb11u1, provided by debian upstream. There is no installable candidate from networkradius.

Now I built deb packages from source myself, which also do not contain rlm_python3.so. Strangely the example module file is available.

**Comment by mcnewton** [comment]
Fixed in 83e193db64f3
